diff -u -r xmonad-contrib-0.14/XMonad/Layout/Fullscreen.hs xmonad-contrib-0.14-test/XMonad/Layout/Fullscreen.hs
--- xmonad-contrib-0.14/XMonad/Layout/Fullscreen.hs	2018-07-31 23:53:27.000000000 +1000
+++ xmonad-contrib-0.14-test/XMonad/Layout/Fullscreen.hs	2018-09-05 09:23:29.590364387 +1000
@@ -32,6 +32,7 @@
 
 import XMonad
 import XMonad.Layout.LayoutModifier
+import qualified XMonad.Util.Rectangle as R
 import XMonad.Util.WindowProperties
 import XMonad.Hooks.ManageHelpers (isFullscreen)
 import qualified XMonad.StackSet as W
@@ -107,9 +108,12 @@
     _ -> Nothing
 
   pureModifier (FullscreenFull frect fulls) rect _ list =
-    (map (flip (,) rect') visfulls ++ rest, Nothing)
-    where visfulls = intersect fulls $ map fst list
-          rest = filter (not . (flip elem visfulls `orP` covers rect')) list
+    (visfulls' ++ rest', Nothing)
+    where (visfulls,rest) = partition (flip elem fulls . fst) list
+          visfulls' = map (second $ const rect') visfulls
+          rest' = if null visfulls'
+                  then rest
+                  else filter (not . R.supersetOf rect' . snd) rest
           rect' = scaleRationalRect rect frect
 
 instance LayoutModifier FullscreenFocus Window where
