diff -u -r status-notifier-item-0.3.0.0/src/StatusNotifier/Util.hs status-notifier-item-0.3.0.0-patch/src/StatusNotifier/Util.hs
--- status-notifier-item-0.3.0.0/src/StatusNotifier/Util.hs	2018-07-06 14:18:16.000000000 +1000
+++ status-notifier-item-0.3.0.0-patch/src/StatusNotifier/Util.hs	2019-01-03 12:36:35.668292967 +1100
@@ -11,6 +11,8 @@
 import           Data.Bits
 import qualified Data.ByteString as BS
 import           Data.Maybe
+import qualified Data.Text as Text
+import qualified Data.Text.IO as Text
 import qualified Data.Vector.Storable as VS
 import           Data.Vector.Storable.ByteString
 import           Data.Word
@@ -24,7 +26,7 @@
 
 getIntrospectionObjectFromFile :: FilePath -> T.ObjectPath -> Q I.Object
 getIntrospectionObjectFromFile filepath nodePath = runIO $
-  head . maybeToList . I.parseXML nodePath <$> readFile filepath
+  head . maybeToList . I.parseXML nodePath <$> Text.readFile filepath
 
 generateClientFromFile :: G.GenerationParams -> Bool -> FilePath -> Q [Dec]
 generateClientFromFile params useObjectPath filepath = do
@@ -122,7 +124,7 @@
   -> T.ObjectPath
   -> IO (Either M.MethodError (Maybe I.Object))
 getInterfaceAt client bus path =
-  right (I.parseXML "/") <$> introspect client bus path
+  right (I.parseXML "/" . Text.pack) <$> introspect client bus path
 
 findM :: Monad m => (a -> m Bool) -> [a] -> m (Maybe a)
 findM p [] = return Nothing
diff -u -r status-notifier-item-0.3.0.0/status-notifier-item.cabal status-notifier-item-0.3.0.0-patch/status-notifier-item.cabal
--- status-notifier-item-0.3.0.0/status-notifier-item.cabal	2018-07-06 14:18:16.000000000 +1000
+++ status-notifier-item-0.3.0.0-patch/status-notifier-item.cabal	2019-01-03 12:22:56.385519736 +1100
@@ -53,6 +53,7 @@
     , network
     , spool >=0.1 && <1.0
     , template-haskell
+    , text
     , transformers
     , vector
   default-language: Haskell2010
